rosbag2_transport/include/rosbag2_transport/readers_wrapper.hpp

```diff
@@ -0,0 +1,131 @@
+// Copyright 2025 Apex.AI, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef ROSBAG2_TRANSPORT__READERS_WRAPPER_HPP_
+#define ROSBAG2_TRANSPORT__READERS_WRAPPER_HPP_
+
+#include <memory>
+#include <utility>
+#include <vector>
+
+#include "rcutils/time.h"
+#include "rosbag2_cpp/bag_events.hpp"
+#include "rosbag2_cpp/reader.hpp"
+#include "rosbag2_storage/serialized_bag_message.hpp"
+#include "rosbag2_storage/storage_options.hpp"
+#include "rosbag2_transport/visibility_control.hpp"
+
+#ifdef _WIN32
+#  pragma warning(push)
+// Suppress warning "rosbag2_transport::ReadersWrapper::pimpl_': class 'std::unique_ptr>'
+// needs to have dll-interface to be used by clients of class 'rosbag2_transport::ReadersWrapper'"
+// Justification:
+// 1. We never inline code in the header that actually calls methods on ReadersWrapperImpl.
+// 2. While the `ReadersWrapperImpl` is defined in the `readers_wrapper_impl.hpp`
+// file, we include it only in the `readers_wrapper.cpp` file, and it does not leak into the
+// external API.
+// 3. The pimpl design pattern imply that implementation details are hidden and shouldn't be
+// exposed with the dll-interface.
+#  pragma warning(disable:4251)
+#endif
+
+
+namespace rosbag2_transport
+{
+class ReadersWrapperImpl;
+
+/// \class ReadersWrapper
+/// \brief This class manages multiple rosbag2_cpp::Reader instances, allowing for
+/// chronological reading of messages across all readers.
+/// \details It maintains a cache of the next message from each reader and provides methods to
+/// retrieve the next message in chronological order, seek to a specific timestamp across all
+/// readers, and apply filters to all readers. It also provides access to the earliest and
+/// latest timestamps across all readers.
+class ROSBAG2_TRANSPORT_PUBLIC ReadersWrapper
```

Comment:
Just a minor suggestion. How about naming the class `ReaderManager` would provide clearer intent, as the class is responsible for managing multiple `rosbag2_cpp::Reader` instances such as opening, seeking, filtering, and reading messages chronologically. IMO "Wrapper" sounds more general pattern and doesn’t convey what the class actually does, while "Manager" directly indicates its role in handling and coordinating the readers?

Reply:
@fujitatomoya Thank you for your review. Your suggestion about renaming to the `ReadersManager` does make sense to me. I will go ahead and rename it.